src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -195,6 +199,13 @@ internal abstract class BrokerClient : IBrokerClient {
             return new RHost(name, callbacks, transport, cts);
         }
 
+        private async Task GetHostInformationAsync() {
```

Comment:
Yes, that's a good idea. You can register a handler and call `CancelPendingRequests`. Problem may happen only if we have more than 1 request

---
In reply to: [79260938](https://github.com/Microsoft/RTVS/pull/2341#discussion_r79260938) [](ancestors = 79260938)